File: backend/schemas/message.py
from pydantic import BaseModel
from typing import Optional, Any, Dict
from datetime import datetime


class MessageBase(BaseModel):
    title: str
    content: str
    message_type: str
    data: Optional[Dict[str, Any]] = None


class MessageCreate(MessageBase):
    receiver_id: int
    sender_id: Optional[int] = None


class MessageUpdate(BaseModel):
    is_read: Optional[bool] = None


class MessageResponse(MessageBase):
    id: int
    sender_id: Optional[int]
    receiver_id: int
    is_read: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class MessageActionBase(BaseModel):
    action_type: str


class MessageActionCreate(MessageActionBase):
    message_id: int


class MessageActionResponse(MessageActionBase):
    id: int
    message_id: int
    user_id: int
    created_at: datetime

    class Config:
        from_attributes = True


class MessageListResponse(BaseModel):
    items: list[MessageResponse]
    total: int
    page: int
    size: int
    pages: int


class FamilyInviteData(BaseModel):
    family_id: int
    family_name: str
    inviter_name: str

File: frontend/src/types/message.ts
export interface Message {
  id: number;
  sender_id?: number;
  receiver_id: number;
  message_type: string;
  title: string;
  content: string;
  data?: Record<string, any>;
  is_read: boolean;
  created_at: string;
  updated_at: string;
}

export interface MessageAction {
  id: number;
  message_id: number;
  user_id: number;
  action_type: string;
  created_at: string;
}

export interface MessageListResponse {
  items: Message[];
  total: number;
  page: number;
  size: number;
  pages: number;
}

export interface MessageCreate {
  receiver_id: number;
  sender_id?: number;
  message_type: string;
  title: string;
  content: string;
  data?: Record<string, any>;
}

export interface MessageUpdate {
  is_read?: boolean;
}

export interface MessageActionCreate {
  message_id: number;
  action_type: string;
}